### PR TITLE
chore: update project structure

### LINKED ISSUE #44: Reorganize deployment resources
Currently the deploy folder contains example deployment resources.

Ideally:

- Helm charts and official deployment resources should go under `deploy/`
- Examples should be under....examples? `deploy/examples`?
- Examples should be simplified (ie. only necessary fields should be part of them for each example)

### PR DESCRIPTION
**Pull request by ramizpolic**
## Overview

Resolves #44 

## Notes for reviewer

Uses new paths for various resources, given as:
```
├── deploy
│   ├── charts - defines helm chart resources
│   ├── defaults - defines resources required to run examples/operator tests
│   ├── dev - defines various development env deployment scripts
│   │   ├── microk8s - example deployments for microk8s env
│   │   └── multi-dc - deployment scripts for running vault instances across multiple clusters for local and AWS envs
│   └── examples - defines example resources for deploying operator
├── hack
│   └── scripts
```

Also, all custom resources in `deploy/examples` dir have been moved to independent files which define complete examples.

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Couple notes:

- I'd much rather split this up into smaller PRs
- deploy/crd is not quite right: those are rather CR examples
- deploy/test move doesn't feel right, especially since it's gonna be rewritten at some point in the style of the vault-secrets-webhook e2e tests
- Can't find deploy/files anywhere yet there are references to it in code

**Comment by ramizpolic** [comment]
In that case, I'd recommend using `config` directory (similar to how kubebuilder does CRD generation) which can be a common parent to all mentioned resources. 
I could split into multiple PRs, but I feel it won't really matter as much since only a couple of tree changes are impactful.

**Comment by ramizpolic** [comment]
> > In that case, I'd recommend using `config` directory (similar to how kubebuilder does CRD generation) which can be a common parent to all the mentioned resources.
> 
> That doesn't feel quite right either. I mean, not for _all_ (eg. tests).
> 
> Here is my recommendation:
> 
> * deploy/charts/...
> * deploy/examples -> CR examples
> * deploy/examples/base (?) -> base resources (rbac) for examples
> * deploy/examples/... -> additional directories for examples with multiple resources (alternatively: put different objects into one yaml file for these examples)

Adding to this, if we were to put `scripts` to `hack/scripts`, we could also then move `tests` to `hack/tests`, which would fit more clearly. I wanted to avoid `deploy` dir since it is used as a transient dir during resource generation.

**Comment by sagikazarmark** [comment]
> Adding to this, if we were to put `scripts` to `hack/scripts`

Sounds good to me


> we could also then move `tests` to `hack/tests`

Let's not touch tests for now. As I said, I wanna move them to e2e (which is another loosly applied pattern out there) once we rewrite them using the e2e framework.

> I wanted to avoid `deploy` dir since it is used as a transient dir during resource generation.

`deploy` is a widely used pattern IMO. I generally prefer it over config.

File: deploy/examples/cr-istio.yaml [hunk 1]
```diff
@@ -209,3 +209,99 @@ spec:
...
+          image: alpine
+          command:
+            [
+              "sh",
+              "-c",
+              "echo $AWS_SECRET_ACCESS_KEY && echo going to sleep... && sleep 10000",
+            ]
+          env:
+            - name: AWS_SECRET_ACCESS_KEY
+              value: vault:secret/data/accounts/aws#AWS_SECRET_ACCESS_KEY
```
Comment (akijakya) [inline deploy/examples/cr-istio.yaml:307]: You could delete this as well, there are similar deployment examples in /test/deploy, also applying this whole file at once would not make sense as we need the Vault instance to be running first, then start the webhook and only after create deployments with such annotations.

File: deploy/examples/cr-istio.yaml [hunk 2]
```diff
@@ -209,3 +209,99 @@ spec:
...
+  name: vault-auth-delegator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:auth-delegator
+subjects:
+  - kind: ServiceAccount
+    name: vault
+    namespace: vault
+
```
Comment (akijakya) [inline deploy/examples/cr-istio.yaml:263]: I think you could leave this out as it is not much different from deploy/default/rbac.yaml

**Review by akijakya — APPROVED** [review]
I left some comments but otherwise LGTM 💯

**Review by akijakya — APPROVED** [review]

File: .dockerignore [hunk 3]
```diff
@@ -3,11 +3,9 @@
...
 /e2e/
-
-/charts/
-/examples/
-/scripts/
 /test/
+/hack/
+/deploy/
+
+/Dockerfile
```
Comment (sagikazarmark) [inline .dockerignore:11]: Please install editorconfig plugin for tailing newlines

File: hack/scripts/validate-config-crud/validate-config-crud.sh [hunk 4]
```diff
@@ -13,7 +13,7 @@ if [ "${log_level,,}" == 'debug' ]; then
 fi
 
 bank_vaults_config_key="${1}"
-bank_vaults_config_file_orig="scripts/validate-config-crud/vault-config.yml"
+bank_vaults_config_file_orig="hack/scripts/validate-config-crud/vault-config.yml"
```
Comment (sagikazarmark) [inline hack/scripts/validate-config-crud/validate-config-crud.sh:16]: This directory can be deleted, it's not used in this project.

File: .dockerignore (hunk 3 above)
Comment (sagikazarmark) [reply .dockerignore:11]:
```suggestion
/deploy/
/Dockerfile
/e2e/
/hack/

/test/

```

File: .yamlignore [hunk 5]
```diff
@@ -1,7 +1,3 @@
-/deploy/
-/e2e/deploy/
-/e2e/test/
-
-/charts/
-/examples/
 /test/
+/e2e/
+/deploy/
```
Comment (sagikazarmark) [inline .yamlignore:3]: Can we restore the original order?

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Couple more nits. Other than those, LGTM.

**Review by sagikazarmark — APPROVED** [review]
LGTM thanks @ramizpolic 

Nice work!

**Pull request merged**